src/databricks/labs/ucx/source_code/notebook_linter.py

```diff
@@ -14,22 +14,35 @@ class NotebookLinter:
     def __init__(self, langs: Languages, notebook: Notebook):
         self._languages: Languages = langs
         self._notebook: Notebook = notebook
+        self._schema: str = DEFAULT_SCHEMA
+
+    @property
+    def schema(self):
+        # This linter is a controller of a number of linters for each cell
+        # language in the notebook. It is not generally useful to know the schema
+        # of the last cell that was linted, except for testing purposes.
+        return self._schema
 
     @classmethod
     def from_source(cls, langs: Languages, source: str, default_language: Language) -> 'NotebookLinter':
         notebook = Notebook.parse("", source, default_language)
```

Comment:
```suggestion
    def from_source(cls, index: MigrationIndex, source: str, default_language: Language) -> 'NotebookLinter':
        languages = Languages(index)
        notebook = Notebook.parse("", source, default_language)
```
this way we ensure that `CurrentSessionState` is regenerated very time we're linting new notebook

Reply:
Done

Reply:
Linter interface is unchanged